_posts/2021-02-11-tabnet/tabnet.Rmd

```diff
@@ -0,0 +1,411 @@
+---
+title: "torch, tidymodels, and high-energy physics"
+description: > 
+ Today we introduce tabnet, a torch implementation of "TabNet: Attentive Interpretable Tabular Learning" that is fully integrated with the tidymodels framework. Per se, already, tabnet was designed to require very little data pre-processing; thanks to tidymodels, hyperparameter tuning (so often cumbersome in deep learning) becomes convenient and even, fun!
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatabnet
+date: 02-11-2021
+bibliography: bibliography.bib
+categories:
+  - Torch
+  - R
+  - Tabular Data
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/d.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+So what's with the clickbait (*high-energy physics*)? Well, it's not just clickbait only. To showcase TabNet, we will be using the [Higgs](https://archive.ics.uci.edu/ml/datasets/HIGGS) dataset (@higgs) available at UCI Machine Learning Repository. I don't know about you, but I always enjoy using datasets that motivate me to learn more about things. But first, let's get acquainted with the main actors of this post!
+
+# TabNet
+
+TabNet was introduced in @arik2020tabnet. It is interesting for three reasons:
+
+-   It claims highly competitive performance on tabular data, an area where deep learning has not gained so much of a reputation yet.
+
+-   It includes interpretability[^1] features by design.
+
+-   It is claimed to significantly profit from self-supervised pre-training, again in an area where this is all but undeserving of mention.
+
+[^1]: I'm using the term "naively" here. For a short discussion, see the final section, [Interpretable, explainable, you tell me -- beyond arbitrary definitions].
+
+In this post, we won't go into (3), but we do expand on (2), the ways TabNet allows access to its inner workings.
+
+How do we use TabNet from R? The `torch` ecosystem includes a package -- `tabnet` -- that not just implements the model of the same name but also, allows to make use of it as part of a `tidymodels` workflow.
```

Comment:
```suggestion
How do we use TabNet from R? The `torch` ecosystem includes a package -- `tabnet` -- that not only implements the model of the same name, but also allows you to make use of it as part of a `tidymodels` workflow.
```

Reply:
It doesn't have to be "you" but there needs to be a subject between "allows" and "to make use of"

Reply:
I see!